cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -402,6 +402,19 @@ private void handleProgramEvent(ProgramRunId programRunId, ProgramRunStatus prog
         recordedRunRecord = appMetadataStore.recordProgramResumed(programRunId, messageIdBytes, resumeTime);
         writeToHeartBeatTable(recordedRunRecord, resumeTime, programHeartbeatTable);
         break;
+      case STOPPING:
+        Map<String, String> notificationProperties = notification.getProperties();
+        long stoppingTsSecs = getTimeSeconds(notificationProperties, ProgramOptionConstants.STOPPING_TIME);
+        if (stoppingTsSecs == -1L) {
+          LOG.warn("Ignore program stopping notification for program {} without {} specified, {}",
+                   programRunId, ProgramOptionConstants.STOPPING_TIME, notification);
+          return;
+        }
+        long terminateTsSecs = getTimeSeconds(notificationProperties, ProgramOptionConstants.TERMINATE_TIME);
```

Comment:
nit: terminateTime

Reply:
Same comment as the other one
```Terence and I were discussing that ideally timestamp should be in millis. Since this timestamp is in seconds we will explicitly suffix it with secs even if it means it deviates from the current naming convention```